Firmware/mmu2_protocol_logic.cpp

```diff
@@ -664,12 +664,12 @@ StepStatus ProtocolLogic::Step() {
         // We are ok, switching to Idle if there is no potential next request planned.
         // But the trouble is we must report a finished command if the previous command has just been finished
         // i.e. only try to find some planned command if we just finished the Idle cycle
-        bool previousCommandFinished = currentState == &command; // @@TODO this is a nasty hack :( 
-        if( ! ActivatePlannedRequest() ){ // if nothing is planned, switch to Idle
+        bool previousCommandFinished = currentScope == Scope::Command; // @@TODO this is a nasty hack :(
+        if (!ActivatePlannedRequest()) {                               // if nothing is planned, switch to Idle
             SwitchToIdle();
         } else {
```

Comment:
We don't want tot call SwitchToIdle() if we are already in Idle. This causes the issue I mentioned on Slack.

`if (!ActivatePlannedRequest() && currentScope != Scope::Idle)` seemed to work for me.

Reply:
thanks for pointing this out, fixed in 2c37f1db